src/Diagram/stateManager/nodeManager/index.ts

```diff
@@ -217,6 +233,53 @@ export class NodeManager {
       this.eventBus.publish(Events.DiagramEvents.RenderRequested);
     }
   };
+  foldUnfoldChildren = () => {
+    const { node, io } = this.state.hover;
+    if (node && !io) {
+      const root = this.state.nodes.find(f => f.id === node.id);
+      const childrenToFold = new Array<Node>();
+      let foundChildren = new Array<Node>();
+      let foundAllChildren = false;
+
+      if (root === undefined) {
+        return;
+      }
+      let newParents = [root];
+      while (!foundAllChildren) {
+        foundChildren = this.findChildren(newParents);
+        newParents = foundChildren;
+        childrenToFold.push(...foundChildren);
+        foundAllChildren = foundChildren.length === 0 ? true : foundAllChildren;
+      } 
+      if (childrenToFold.length > 0) {
+        childrenToFold.forEach(child => {
+          let node = this.state.nodes.find(f => f.id === child.id);
```

Comment:
what about using `child.hidden` instead of finding a node in the list, does it not work?

Reply:
I used this.state.nodes to change their properties values in context. 
Using child.id I can find unique element in list. If i use child.hidden I will get many elements. As a result it can be a case when I always get the same first hidden element in this.state.nodes.